src/dsp_tools/commands/xmllib/models/resource.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
+
+import warnings
+from collections import defaultdict
+from dataclasses import dataclass
+from dataclasses import field
+
+from lxml import etree
+
+from dsp_tools.commands.xmllib.models.values import IntValue
+from dsp_tools.commands.xmllib.models.values import LinkValue
+from dsp_tools.commands.xmllib.models.values import SimpleText
+from dsp_tools.commands.xmllib.models.values import Value
+from dsp_tools.commands.xmllib.utils import is_string
+from dsp_tools.models.custom_warnings import DspToolsUserWarning
+from dsp_tools.models.exceptions import InputError
+from dsp_tools.utils.uri_util import is_iiif_uri
+
+XML_NAMESPACE_MAP = {None: "https://dasch.swiss/schema", "xsi": "http://www.w3.org/2001/XMLSchema-instance"}
+DASCH_SCHEMA = "{https://dasch.swiss/schema}"
+
+
+@dataclass
+class ResourceCollection:
+    resources: dict[str, Resource] = field(default_factory=dict)
+
+    def to_list(self) -> list[Resource]:
+        return list(self.resources.values())
+
+    def with_Resource(self, resource: Resource) -> ResourceCollection:
```

Comment:
```suggestion
    def with_resource(self, resource: Resource) -> ResourceCollection:
```
python naming conventions :)